rfcs/convergence/make-overrides.md

```diff
@@ -0,0 +1,194 @@
+# RFC: Replace `makeStyles()` implementation / API updates
+
+<!--
+An RFC can be anything. A question, a suggestion, a plan. The purpose of this template is to give some structure to help folks write successful RFCs. However, don't feel constrained by this template; use your best judgement.
+
+Tips for writing a successful RFC:
+
+- Simple plain words that make your point, fancy words obfuscate
+- Try to stay concise, but don't gloss over important details
+- Try to write a neutral problem statement, not one that motivates your desired solution
+- Remember, "Writing is thinking". It's natural to realize new ideas while writing your proposal
+-->
+
+---
+
+@layershifter
+
+## Summary
+
+This RFC proposes to replace existing `makeStyles()` implementation with a new one, i.e. proposes API changes.
+
+## Background
+
+In [microsoft/fluentui#16082](https://github.com/microsoft/fluentui/pull/16082) we have implemented a version of `makeStyles()` that uses atomic classes. Recent work on `makeOverrides()` shows significant performance improvements [microsoft/fluentui#16923](https://github.com/microsoft/fluentui/pull/16923) (around 15%) by changes in the implementation design.
+
+I will use `makeOverrides()` and `makeStyles()` for naming in this RFC to avoid confusion, but a proposal is to replace the current `makeStyles()` implementation with `makeOverrides()`.
```

Comment:
```suggestion
To avoid confusion, the names `makeOverrides()` and `makeStyles` are both used in this RFC. This RFC also proposes the replacement of `makeStyles` with the `makeOverrides` implementation.
```